emcc.py

```diff
@@ -2274,6 +2274,8 @@ def phase_linker_setup(options, state, newargs):
   if settings.USE_PTHREADS:
     setup_pthreads(target)
     settings.JS_LIBRARIES.append((0, 'library_pthread.js'))
+    if settings.PROXY_TO_PTHREAD:
+      settings.PTHREAD_POOL_SIZE_STRICT = 0
```

Comment:
Even in proxy-to-pthread there is some risk, since we ask the main thread to create the worker atm, so if it is blocking on this thread we'd deadlock. Seems safer to leave the warnings?

Reply:
We already didn't show those warnings FWIW so this is not really a change in functionality, just moving it earlier in the process to make it easier to write conditionals.

Reply:
The reason we didn't show the warning in the first place is because yes, we ask the main thread to create the worker via `.postMessage`, and `postMessage` works fine synchronously and allows us to block the thread immediately.

However, `new Worker` (in browsers) works asynchronously and needs an event loop to even start. That's why doing `new Worker` on the same thread is dangerous and results in deadlock if we immediately block, but asking the main thread from `PROXY_TO_PTHREAD` to create a Worker for us, is not.

Reply:
When you clobber a setting like this I like to do one of two things:

1.  Use default_setting helper function to only set if user didn't specify it
2. Error out if user_settings contains this settting with a nice error.  Something like `PTHREAD_POOL_SIZE_STRICT is not compatible with PROXY_TO_PTHREAD` .. or maybe `X is ignored when Y is set`?  

There should be plenty of examples of both in this file I think.